CloudFormation/cf_template.yml

```diff
@@ -76,4 +76,37 @@ Resources:
                   - 's3:PutObject'
                 Resource:
                   - 'arn:aws:s3:::my-source-bucket'
```

Comment:
have you tested if this works? I dont think it will work. here you creating a lambda execution role, giving lambda the access to two buckets, named my-source-bucket and 'my-destination-bucket'. i dont see these buckets created anywhere.
you did create 'datalake-source-s3-mz' and 'datalake-raw-s3-mz'. maybe this is what you mean?

also, it is better if you use intrinsic functions to construct the s3 arn or retrieve it from the resource directly. See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html for all intrinsic functions. see this cheatsheet for what function to use to retrieve s3 arn:
https://theburningmonk.com/cloudformation-ref-and-getatt-cheatsheet/
Search for AWS::S3::Bucket